applications/ContactStructuralMechanicsApplication/custom_utilities/contact_utilities.h

```diff
@@ -238,40 +238,38 @@ class ContactUtilities
         const int num_conditions = static_cast<int>(conditions_array.size());
         
         #pragma omp parallel for
-        for(int i = 0; i < num_conditions; i++) 
+        for(int i = 0; i < num_conditions; ++i) 
         {
             auto it_cond = conditions_array.begin() + i;
+            GeometryType& this_geometry = it_cond->GetGeometry();
             
-            if (it_cond->Is(SLAVE) || it_cond->Is(MASTER) || it_cond->Is(ACTIVE))
+            aux_coords = this_geometry.PointLocalCoordinates(aux_coords, this_geometry.Center());
+            it_cond->SetValue(NORMAL, this_geometry.Normal(aux_coords));
+            const array_1d<double, 3>& normal = it_cond->GetValue(NORMAL);
+            
+            const unsigned int number_nodes = this_geometry.PointsNumber();
+            const double & rArea = this_geometry.Area()/number_nodes;
+            
+            for (unsigned int i = 0; i < number_nodes; ++i)
             {
-                aux_coords = it_cond->GetGeometry().PointLocalCoordinates(aux_coords, it_cond->GetGeometry().Center());
-                array_1d<double, 3>& rNormal = it_cond->GetValue(NORMAL);
-                rNormal = it_cond->GetGeometry().Normal(aux_coords);
-                
-                const unsigned int number_nodes = it_cond->GetGeometry().PointsNumber();
-                const double & rArea = it_cond->GetGeometry().Area()/number_nodes;
-                
-                for (unsigned int i = 0; i < number_nodes; i++)
+                auto& this_node = this_geometry[i];
+                #pragma omp atomic
```

Comment:
i think you first needs to get a reference, and then to do it atomic, smthg like

      double& nodal_area = this_node.GetValue(NODAL_AREA)
      #pragma omp atomic 
      nodal_area += ...